lib/matplotlib/axes/_axes.py

```diff
@@ -4636,6 +4644,7 @@ def invalid_shape_exception(csize, xsize):
             colors = None  # use cmap, norm after collection is created
         return c, colors, edgecolors
 
+    @_api.make_keyword_only("3.9", "marker")
```

Comment:
I can see an arguament for pushing this one deeper

```python
ax.scatter(x, y, s, c, 'o')
```

scans reasonably well, but I suspect is rare.